app/views.py

```diff
@@ -19,19 +19,45 @@ class IndexView(TemplateView):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        if "type" not in self.request.GET or "q" not in self.request.GET:
+
+        context["type"] = ["challenge", "project"]
+
+        context["tags"] = Tag.objects.all()
+        context["selected_tags"] = []
+
+        if "type" not in self.request.GET or (
+            "q" not in self.request.GET and "tag" not in self.request.GET
+        ):
             context["challenges"] = Challenge.objects.all()[:3]
             context["projects"] = Project.objects.all()[:9]
-            return context
-        query = self.request.GET["q"]
-        if self.request.GET["type"] == "challenge":
-            context["challenges"] = Challenge.objects.all().filter(
-                Q(name__contains=query) | Q(description__contains=query)
-            )
-        elif self.request.GET["type"] == "project":
-            context["projects"] = Project.objects.all().filter(
-                Q(name__contains=query) | Q(description__contains=query)
-            )
+        else:
+            initiative = self.request.GET["type"]
+            context["type"] = [initiative]
+
+            if initiative == "challenge":
+                queryset = Challenge.objects.all()
+
+            elif initiative == "project":
+                queryset = Project.objects.all()
```

Comment:
We need to do something if it's neither of these.

Reply:
Perhaps we could just ignore it and resort to displaying the unqueried `explore` page (i.e. when checking if a `tag` parameter exists, also verify its value to be either `"project"` or `"challenge"`)?

Alternative could be an error page

Reply:
Error page is too much work, just assume it's absent.

Reply:
So either redirect to unqueried or return unqueried.

Reply:
(8ba3a29846c08b2f852257a477b0426fa98f9deb)